src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,35 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        StringBuilder builder = new StringBuilder();
+        int[] empolyerSalaryPeriod = new int[names.length];
+        String[] dataUser;
+        LocalDate firstDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate lastDate = LocalDate.parse(dateTo, formatter);
+        LocalDate currentDate;
+        for (String dt : data) {
+            dataUser = dt.split(" ");
+            currentDate = LocalDate.parse(dataUser[0], formatter);
+            if (currentDate.isAfter(firstDate) && currentDate.isBefore(lastDate.plusDays(1))) {
+                for (int j = 0; j < names.length; j++) {
+                    if (names[j].equals(dataUser[1])) {
+                        empolyerSalaryPeriod[j] += Integer.parseInt(dataUser[2])
+                                * Integer.parseInt(dataUser[3]);
```

Comment:
Any magic numbers should be constants (https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)
Your code should be easy to read. Move all hardcoded values to constant fields and give them informative names.
Separator " - " can be constant too.